src/pybind/mgr/progress/module.py

```diff
@@ -541,11 +541,10 @@ def notify(self, notify_type, notify_data):
             ))
             self._osdmap_changed(old_osdmap, self._latest_osdmap)
         elif notify_type == "pg_summary":
-            data = self.get("pg_dump")
             for ev_id in list(self._events):
                 ev = self._events[ev_id]
                 if isinstance(ev, PgRecoveryEvent):
-                    ev.pg_update(data, self.log)
+                    ev.pg_update(self.get("pg_stats"), self.log)
```

Comment:
it'd be more efficient to keep this outside the loop - there could be many events

Reply:
fixed